src/migrate/Migrator.js

```diff
@@ -419,15 +440,42 @@ export default class Migrator {
   }
 
   setConfig(config) {
-    return assign({}, CONFIG_DEFAULT, this.config || {}, config);
+    const newConfig = assign({}, CONFIG_DEFAULT, this.config || {}, config);
+
+    // If migrationSource is not specified, but fs related config has been specified
+    // recreate the FsMigration source so the new filesystem related settings
+    // are used
+    if (
+      config &&
+      !config.migrationSource &&
```

Comment:
I think this is the only useful check and the logic should be as following: "if newConfig.migrationSource -> noop, else newConfig.migrationSource = new FSMigrationSource". Everything else doesn't seem to result in any meaningful branches.

Reply:
But then this wouldn't work

```
const migrator = new Migrator({ directory: './foo' })
migrator.latest({ directory: './bar' })
```

Because the ctor would initialise `migrationSource`, then the call to latest would check to see if newConfig.migrationSource is set. It is, so no-op. Leaving the FsMigrationSource looking at `./foo` rather than `./bar`.

Reply:
Ah, OK, now your logic makes sense. Could you clarify comment a little bit to make it more explicit that you are accounting for the case where migrationSource configuration has changed between previous config and newly passed one?

Reply:
I could add some specific tests around it. Was confusing to write too :P

Reply:
Added some comments to clarify